python/tk_multi_publish2/dialog.py

```diff
@@ -1478,6 +1484,38 @@ def _show_no_items_error(self):
         self.ui.main_stack.setCurrentIndex(self.PUBLISH_SCREEN)
         self._overlay.show_no_items_error()
 
+    def _validate_task_required(self):
+        """
+        Validates that a task is selected for every item and disables/enables the
+        validate and publish buttons and finally change the color for the task
+        label.
+        """
+        # Avoid validation if the setting `task_required` is False or not exists
+        if not self._bundle.get_setting("task_required"):
+            return
+
+        all_items_selected_task = True
+        for context_index in xrange(self.ui.items_tree.topLevelItemCount()):
+            context_item = self.ui.items_tree.topLevelItem(context_index)
+
+            if hasattr(context_item, "context") and not context_item.context.task:
+                all_items_selected_task = False
+                break
+
+        if not all_items_selected_task:
+            # disable buttons
+            self.ui.publish.setEnabled(False)
+            self.ui.validate.setEnabled(False)
+            # change task label color to RED
+            self.ui.context_widget.ui.task_label.setStyleSheet("color: red")
```

Comment:
See [docs here](https://developer.shotgunsoftware.com/tk-core/platform.html#sgtk.platform.Engine.style_constants). I would remove the hardcoded style directive from the code, put it into the stylesheet file and use `{{SG_ALERT_COLOR}}`.